packages/components/src/custom-select-control/index.js

```diff
@@ -58,25 +64,23 @@ export default function CustomSelectControl( {
 	label,
 	options: items,
 	onChange: onSelectedItemChange,
-	value: _selectedItem,
+	value,
 } ) {
+	const valueIndex = items.findIndex( ( item ) => item.key === value.key );
```

Comment:
This will throw if the input is not controlled. `value` wouldn't be defined.

Reply:
Whoopsies, forgot about that!